MantidPlot/src/Graph3D.cpp

```diff
@@ -2483,120 +2494,437 @@ Graph3D::~Graph3D() {
+
+  int xCol = w->colIndex(xColName);
+  int yCol = w->colIndex(yColName);
+  int zCol = w->colIndex(zColName);
+
+  loadData(w, xCol, yCol, zCol, params.xStart, params.xStop, params.yStart,
+           params.yStop, params.zStart, params.zStop);
+
+  QString label = caption;
+  if (app->alreadyUsedName(label))
+    label = app->generateUniqueName("Graph");
+
+  setWindowTitle(label);
+  setName(label);
+  app->initPlot3D(this);
+}
+
+void Graph3D::setupPlotParametricSurface(ApplicationWindow *app,
+                                         const SurfaceFunctionParams &params) {
+  QString label = app->generateUniqueName("Graph");
+
+  resize(500, 400);
+  setWindowTitle(label);
+  setName(label);
+  app->customPlot3D(this);
+  addParametricSurface(QString::fromStdString(params.xFormula),
+                       QString::fromStdString(params.yFormula),
+                       QString::fromStdString(params.zFormula), params.uStart,
+                       params.uEnd, params.vStart, params.vEnd, params.columns,
+                       params.rows, params.uPeriodic, params.vPeriodic);
+  app->initPlot3D(this);
+}
+
+void Graph3D::setupPlotSurface(ApplicationWindow *app,
+                               const SurfaceFunctionParams &params) {
+
+  QString label = app->generateUniqueName("Graph");
+
+  resize(500, 400);
+  setWindowTitle(label);
+  setName(label);
+  app->customPlot3D(this);
+  addFunction(QString::fromStdString(params.xFormula), params.xStart,
+              params.xStop, params.yStart, params.yStop, params.zStart,
+              params.zStop, params.columns, params.rows);
+
+  app->initPlot3D(this);
+}
+
+void Graph3D::setupMatrixPlot3D(ApplicationWindow *app, const QString &caption,
+                                const SurfaceFunctionParams &params) {
+
+  QString name = QString::fromStdString(params.formula);
+  name.remove("matrix<", Qt::CaseSensitive);
+  name.remove(">", Qt::CaseSensitive);
+  Matrix *m = app->matrix(name);
+  if (!m)
+    return;
+
+  setWindowTitle(caption);
+  setName(caption);
+  addMatrixData(m, params.xStart, params.xStop, params.yStart, params.yStop,
+                params.zStart, params.zStop);
+  update();
+
+  app->initPlot3D(this);
+}
+
+void Graph3D::setupMantidMatrixPlot3D(ApplicationWindow *app,
+                                      TSVSerialiser &tsv) {
+  using MantidQt::API::PlotAxis;
+  MantidMatrix *matrix = readWorkspaceForPlot(app, tsv);
+  int style = read3DPlotStyle(tsv);
+
+  if (matrix) {
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+    QString labl = app->generateUniqueName("Graph");
+
+    resize(500, 400);
+    setWindowTitle(labl);
+    setName(labl);
+    setTitle("Workspace " + matrix->name());
+    app->customPlot3D(this);
+    customPlotStyle(style);
+    int resCol = matrix->numCols() / 200;
+    int resRow = matrix->numRows() / 200;
+    setResolution(qMax(resCol, resRow));
+
+    double zMin = 1e300;
+    double zMax = -1e300;
```

Comment:
It might be worth using [`numeric_limits`](http://en.cppreference.com/w/cpp/types/numeric_limits) instead of hard-coded values